runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c

```diff
@@ -119,12 +121,22 @@ static void genNumLocalesOptions(FILE* slurmFile, sbatchVersion sbatch,
     walltime = getenv("CHPL_LAUNCHER_WALLTIME");
   }
 
+  // command line partition takes precedence over env var
+  if (!partition) {
+    partition = getenv("SALLOC_PARTITION");
```

Comment:
I don't think it makes any sense to be checking slurm env vars. Similar to the line above where we're checking `CHPL_LAUNCHER_WALLTIME` instead of `SLURM_TIMELIMIT`: if a user set `SALLOC_PARTITION`, there's no reason for us to be throwing `--partition`. I think this should turn into checking if `CHPL_LAUNCHER_PARTITION` is set instead.